Align agg_id with role. (#335)
ietf-wg-ppm/draft-ietf-ppm-dap: draft-ietf-ppm-dap.md

~~~

`vdaf_verify_key` is the VDAF verification key shared by the aggregators;
`agg_id` is the aggregator ID, equal to the role in {{protocol-definition}}

> (`0x02` for the Leader and `0x03` for the helper);

Reviewer: This would cause problems given the current text of draft-irtf-cfrg-vdaf. There, `agg_id` is expected to be less than the total number of shares. The pseudocode also hinges some decisions on whether `agg_id == 0`. To simplify this, we either need to subtract two here, renumber the Role enum above, or coordinate some larger rearrangement with a future VDAF draft.

Reviewer 2: ha, just saw this :)

Reviewer 3: It seems bad that an arbitrary change in DAP breaks VDAF. Let's do whatever is needed to decouple these things so that the aggregator ID can be freely assigned.

Reviewer 2: In general the computations of each of the aggregator's aren't identical. For example, for Prio3 the first aggregator's (agg_id==0) is larger than the other's. On the other hand, for Poplar1 they're all roughly the same size. In general there is no reason to think that there will always be a notion of a "leader share".

Do you have thoughts about how to clean this up? David suggested one fix. An alternative would be to re-label the aggregators here.

Author: Oh, indeed it is being used basically as an array index to identify each aggregator's data.

I have no strong opinion.
This was based on the desire to expand to more aggregators later. We could also use 0 = leader, 1 = helper (and 2, 3, .. if necessary) 254 = collector, 255 = client and change role accordingly?

Or just leave everything as is, though I was equally surprised as Sean about the discrepancy.

Reviewer 2: I like you're suggestion for relabeling Simon, let's go with that and see what it looks like. Note that you'll have to propagate the changes to wherever we HPKE encrypt or decrypt something.

Author: ISTM @chris-wood is advocating for just leaving it as is. We can but we somehow need to specify how to convert the arbitrary IDs from the DAP spec to continues numbers with leader=0 for the VDAF spec.

> `agg_id` is the aggregator ID, equal to the role in {{protocol-definition}}
> (`0x02` for the Leader and `0x03` for the helper);

Reviewer 2: This is a breaking change for VDAF. We could either:
1. Update the agg ID semantics in VDAF and cut a draft
2. Align server role with agg ID

Option (2.) might look like this:

```suggestion
`agg_id` is the aggregator ID, equal to `server_role - 0x02`, where `server_role` is the aggregators' role in {{protocol-definition}};
```
Alternatively, we can re-assign the server roles to align with the VDAF draft.

Reviewer 2: Duplicates @divergentdave's comment.
